src/zarr/api/asynchronous.py

```diff
@@ -198,7 +197,7 @@ async def open(
     path: str | None = None,
     storage_options: dict[str, Any] | None = None,
     **kwargs: Any,  # TODO: type kwargs as valid args to open_array
-) -> AsyncArray | AsyncGroup:
+) -> AsyncArray[ArrayV2Metadata] | AsyncArray[ArrayV3Metadata] | AsyncGroup:
```

Comment:
This question applies here and everywhere else you use the v2/v3 Metadata types. 

Since the generic only includes two variants (`ArrayV2Metadata` and `ArrayV3Metadata`), what value do we get by using both together here (as apposed to simply using the `AsyncArray` type?

Reply:
that is purely to silence mypy complaining about a generic missing its type parameter. If there's a cleaner way to annotate it then I'll do it. Would `AsyncArray[Any]` work there? (I can play around with this locally)